### DISCUSSION TITLE
warning/prevention of/when deleting media that is in use

### DISCUSSION THREAD

**Discussion by olarsson** [payloadcms/payload#3865] [Feature Requests & Ideas]
it would be very good if there was a way to warn users when deleting a media that is already in use somewhere in the database. for example a modal that shows where its being referenced and the possibility of warning or preventing deletion of an in use media all together would be a great feature. mainly to stop clients from messing things up in the frontend.

**Comment by mvdve** [comment]
What you can do is create a `beforeDelete` hook and throw an error when the media item is in use. A generic solution would be quite difficult because the media field can be used all over the place (collections, globals, blocks...).

The thrown error will be shown in de default message component at the bottom of the page.
